product_optional_product_quantity/models/product_template.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    product_optional_line_ids = fields.One2many(
+        string="Optional Products",
+        comodel_name="product.optional.line",
+        inverse_name="product_tmpl_id",
+    )
+    optional_product_ids = fields.Many2many(
+        compute="_compute_optional_product_ids",
+        readonly=False,
+        store=True,
+    )
+
+    @api.depends("product_optional_line_ids")
+    def _compute_optional_product_ids(self):
+        """
+        To avoid possible issues with optional products in other modules,
+        we need to compute them based on optional product lines, because
+        we hide the original field.
+        """
+        optional_quantity_enabled = self.env.user.has_group(
+            "product_optional_product_quantity.group_product_optional_quantity"
+        )
+        for r in self:
+            if optional_quantity_enabled:
+                r.optional_product_ids = r.product_optional_line_ids.mapped(
+                    "optional_product_tmpl_id"
+                )
+            else:
+                r.optional_product_ids = r.optional_product_ids
```

Comment:
The group is specifically assigned to the quantity field. However, in this scenario, why aren't the **optional_product_ids** being populated?